build.gradle

```diff
@@ -95,6 +95,64 @@ spotbugsTest {
 java.sourceCompatibility = JavaVersion.VERSION_11
 java.targetCompatibility = JavaVersion.VERSION_11
 
+
+compileJava {
+    options.compilerArgs = [
+            '-Xlint:auxiliaryclass',
+            '-Xlint:cast',
+            '-Xlint:classfile',
+            '-Xlint:dep-ann',
+            '-Xlint:divzero',
+            '-Xlint:empty',
+            '-Xlint:exports',
+            '-Xlint:fallthrough',
+            '-Xlint:finally',
+            '-Xlint:module',
+            '-Xlint:opens',
+            '-Xlint:overloads',
+            '-Xlint:overrides',
+            '-Xlint:-processing',
```

Comment:
Why ignore these warnings?  Could we allow warnings on everything instead of adding these different types one at at time?

Reply:
Well I took it from OpenSearch build. And the rule in this case simple:
 - `-Xall` - validate everything
 - [`-X...`,`-X...`,`-X...`...] - validate list of what I want to validate

Reply:
I mean I'd prefer if we just used `-Xall`, is there any reason we can't or a specific reason to block this processing linter entries?

Reply:
TBH If we do not have rules blindly follow OpenSearch core rules for build, I'm for that.

Reply:
If we blanket copying them from OS, could you include a path to find them on line 98 so we can regenerate if desired?